cmd/syncer/options/options.go

```diff
@@ -52,7 +53,7 @@ func (options *Options) AddFlags(fs *pflag.FlagSet) {
 	fs.StringVar(&options.PclusterID, "workload-cluster-name", options.PclusterID,
 		fmt.Sprintf("ID of the -to cluster. Resources with this ID set in the '%s' label will be synced.", nscontroller.ClusterLabel))
 	fs.StringArrayVarP(&options.SyncedResourceTypes, "sync-resources", "r", options.SyncedResourceTypes, "Resources to be synchronized in kcp.")
-
+	fs.StringVar(&options.KCPHost, "kcp-host", options.KCPHost, "Hostname of the kcp server (url:port) if not set, defaults to the FromKubeConfig host.")
```

Comment:
is this format validated? Does url imply `https://`?

full-stop after the parenthesis.

Reply:
why do we want to have an override of the kubeconfig host field and not set it correctly the first time?

Reply:
When the Syncer is in push mode, the From address is "localhost", that's not going to work when trying to connect back to KCP from a downstream POD.

Reply:
then this is my gap, I was thinking that in push mode the user pass the kubeconfig file to the syncer ... how is this kubeconfig passed to the syncer then? 
The key is to know how this  kubeconfig is generated and why we can not pass the right url there :)

Reply:
FYI we're not intending to support 'push mode' going forward.

Reply:
I use to mix both modes, I'm trying to say the one that the syncer connects to kcp, is that pull?

Reply:
- 'push': the syncer for a given cluster runs as part of the kcp server process
- 'pull': the syncer for a given cluster runs external to kcp

We're not intending for kcp to keep running syncers because we don't want to depend on workload clusters being network accessible to kcp.

Reply:
> When the Syncer is in push mode, the From address is "localhost", that's not going to work when trying to connect back to KCP from a downstream POD.

if the pod has hostNetwork:true it will work

Reply:
So syncer will need to run with `hostNetwork` to function? 😄

In seriousness, I don't think we want this arg. The kcp kubeconfig will need to be valid for a syncer running out-of-process.

Reply:
I agree, we can get rid of this Arg, what do you think, if instead of setting 127.0.0.1 by default on the kubeconfig used by the syncer, we use the Binding address?

Reply:
I think that's fine for this PR. And whatever mechanism creates syncer yaml will need to know how to set that externally-accessible address.